### PR TITLE
docs: updated algolia guide with new settings

### PR DESCRIPTION
**Pull request by shahednasser**
Updated algolia's configuration based on changes in the upcoming latest version.

Closes DOCS-633

### CODE REVIEW

File: docs/content/plugins/search/algolia.md [hunk 1]
```diff
@@ -100,38 +100,94 @@ Where `<YOUR_APP_ID>` and `<YOUR_ADMIN_API_KEY>` are respectively the Applicatio
...
 ```
 
-The `searchableAttributes` are the attributes in a product that are searchable, and `attributesToRetrieve` are the attributes to retrieve for each product result. You’re free to make changes to these attributes as you see fit, but these are the recommended attributes.
+Where:
+
+- `indexName`: the name of the index to create in Algolia. For example, `products`. Its value is an object containing the following properties:
+  - `indexSettings`: an object that includes the following properties:
+    - `searchableAttributes`: an array of strings indicating the attributes in the product entity that can be searched.
+    - `attributesToRetrieve`: an array of strings indicating the attributes in the product entity that should be retrieved in the search results.
+  - `transformer`: an optional function that accepts a product as a parameter and returns an object to be indexed. This allows you to have more control over what you're indexing. For example, you can add details related to variants or custom relations, or you can filter out certain products.
```
Comment (shahednasser) [inline docs/content/plugins/search/algolia.md:150]: @olivermrbl can you please confirm my understanding of the `transform` function?

Comment (olivermrbl) [reply docs/content/plugins/search/algolia.md:150]: Yes, this is correct. You can use it for any type of Medusa entity, you wish to index. But right now, we only supports products :)

File: docs/content/plugins/search/algolia.md [hunk 2]
```diff
@@ -100,38 +100,94 @@ Where `<YOUR_APP_ID>` and `<YOUR_ADMIN_API_KEY>` are respectively the Applicatio
...
 ]
 ```
 
-The `searchableAttributes` are the attributes in a product that are searchable, and `attributesToRetrieve` are the attributes to retrieve for each product result. You’re free to make changes to these attributes as you see fit, but these are the recommended attributes.
+Where:
+
+- `indexName`: the name of the index to create in Algolia. For example, `products`. Its value is an object containing the following properties:
+  - `indexSettings`: an object that includes the following properties:
+    - `searchableAttributes`: an array of strings indicating the attributes in the product entity that can be searched.
+    - `attributesToRetrieve`: an array of strings indicating the attributes in the product entity that should be retrieved in the search results.
```
Comment (olivermrbl) [inline docs/content/plugins/search/algolia.md:149]: It should be noted, that these are just examples of settings passed to the provider. Anything you specify in `indexSettings` will be passed to `setSettings` from Algolia's SDK.

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**